pymongo/client_session.py

```diff
@@ -1111,7 +1120,7 @@ def get_server_session(self, session_timeout_minutes: float) -> _ServerSession:
             if not s.timed_out(session_timeout_minutes):
                 return s
 
-        return _ServerSession(self.generation)
+        return _ServerSession(self.generation, old.session_id)
```

Comment:
This is odd. If the cache is empty we'll create a new session with the same id but if the cache isn't empty we'll return an existing session with a different id. What problem are we trying to solve here?

Reply:
Hmm, good point if the cache is not empty. Our unified test runner expects that sessions have an id on creation, with several tests relying on that behavior. Adding a simple ping command after creation to populate the session id emits unexpected events that cause those same tests to fail.

Reply:
Oh I see. I think the only approach is that accessing the session_id property needs to materialize the session too.

Reply:
Sorry, I don't quite understand what you mean. You're saying we need to have the session materialize when `session_id` is accessed, but without actually checking if sessions are supported?

Reply:
Yes. The session support check would still happen later in _apply_to.